routes/volunteers_routes.js

```diff
@@ -0,0 +1,217 @@
+var volunteers_model = require('../models/volunteers');
+var Department = volunteers_model.Department;
+var Role = volunteers_model.VolunteerRole;
+var Volunteer = volunteers_model.Volunteer;
+const userRole = require('../libs/user_role');
+var DrupalAccess = require('../libs/drupal_acces').DrupalAccess;
+var _ = require('lodash');
+var logger = require('../libs/logger');
+
+//roles ... consider moving to some other file.
+const VOLUNTEER_MANAGER = 0;
+const VOLUNTEER_DEPT_MANAGER = 2;
+
+//GET /volunteer/departments
+var get_departments = function(req, res) {
+    Department.fetchAll().then(function(deps) {
+        res.json(deps.toJSON());
+    }).catch((err) => {
+        res.status(500).json({
+            error: true,
+            data: {
+                message: err.message
+            }
+        });
+    });
+};
+//GET /volunteer/volunteers
+var get_roles = function(req, res) {
+    Role.fetchAll().then(function(deps) {
+        res.json(deps.toJSON());
+    }).catch((err) => {
+        res.status(500).json({
+            error: true,
+            data: {
+                message: err.message
+            }
+        });
+    });
+}
+
+function __has_permissions(user_id, perm_level, next, err) {
+    return Volunteer.get_by_user(user_id).then((vol_data) => {
+        if (vol_data.get('role_id') <= perm_level) {
+            next();
+        } else {
+            err('No Permissions');
+        }
+    });
+};
+//GET /volunteer/volunteers
+var get_volunteers = function(req, res) {
+    __has_permissions(req.user.id, VOLUNTEER_MANAGER, () => {
+        DrupalAccess.get_user_by_email(req.query.email).then((user) => {
+            Volunteer.query((qb) => {
+                qb.where('user_id', user.id);
```

Comment:
basically it will return all volunteering data for a given user (each user can volunteer in several departments). The spec, requests that we will be able to search volunteers by name ans last name' but after talking to @may-ben-arie I understood that search by email (which supposed to be unique in the users table) will suffice.